src/app/util/CHIPDeviceCallbacksMgr.cpp

```diff
@@ -64,8 +64,22 @@ CHIP_ERROR CHIPDeviceCallbacksMgr::AddResponseCallback(NodeId nodeId, uint8_t se
     VerifyOrReturnError(onFailureCallback != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
 
     ResponseCallbackInfo info = { nodeId, sequenceNumber };
-    memcpy(&onSuccessCallback->mInfoPtr, &info, sizeof(info));
-    memcpy(&onFailureCallback->mInfoPtr, &info, sizeof(info));
+
+    onSuccessCallback->mInfoPtr = CHIPPlatformMemoryAlloc(sizeof(info));
```

Comment:
should we validate onSuccessCallback-mInfoPtr == nullptr to avoid memory leaks before allocating (thinking what if the same callback is used/set multiple times?).

Where is this memory freed?

Reply:
From what I read in the previous code, it is storing it into the pointer storage itself .. which may be ok as long as one `static_assert(sizeof(mInfoPtr) >= sizeof(info))`

Reply:
@vivien-apple ?

Reply:
This code is now allocating specific info per callback type, instead of using the info in-place, so it's OK w/o static_assert.

Reply:
New code is ok without an assert, however it is harder to follow lifetime.

I am saying old code could have been correct with an assert (although it is I think working by chance since the assert would not pass). I would like to improve on the new code to somehow make object lifetime obviously correct.